### PR TITLE
Fixes Bayesian ANOVAs

### PR DESCRIPTION
**Pull request by vandenman**
Fixes:
https://github.com/jasp-stats/jasp-test-release/issues/111 
https://github.com/jasp-stats/jasp-test-release/issues/112
https://github.com/jasp-stats/jasp-test-release/issues/140
https://github.com/jasp-stats/jasp-test-release/issues/144
https://github.com/jasp-stats/jasp-test-release/issues/145
https://github.com/jasp-stats/jasp-test-release/issues/146

There are also some other bugfixes in here that were not picked up during testing (e.g., the prior of the covariates, `priorCovariates`, didn't work for repeated measures).  

@timkdj Did I modify the doubleField in the qml in the intended way?

### CODE REVIEW

**Comment by TimKDJ** [comment]
The textfield clipping seems to be an issue for textfields across analyses, so it would be good if we can fix this a bit more generally. Hopefully @boutinb can figure out what causes this, as I cannot reproduce the issue on OSX. Do the doublefields clip for you, @vandenman, when you do not set the fieldwidth?

**Comment by vandenman** [comment]
I'm getting this without the fieldwidth:
afbeelding

versus the old one:
afbeelding

so it looks like removing fieldwidth just does the trick!

File: JASP-Engine/JASP/R/commonAnovaBayesian.R [hunk 1]
```diff
@@ -673,85 +771,92 @@
...
-        distMin <- which(stats::dist(dfCri$xmin) < tol)
-        distMax <- which(stats::dist(dfCri$xmin) < tol)
-        distBoth <- intersect(distMin, distMax)
-        if (length(distBoth) > 0) {
-          aesLine <- ggplot2::aes(x = x, y = y, g = g, color = g, linetype = g)
-          aesErrorbar <- ggplot2::aes(xmin = xmin, xmax = xmax, y = y, group = g, color = g, linetype = g)
+      if (all(c(densities[, ind, "y"]) == 0)) { # only TRUE is never sampled and set to 0
+        plot <- createJaspPlot(title = nms[i], width = 400, height = 400)
+        plot$setError(paste("Could not plot posterior distribution.",
+                            "Check if an error occured for models including this parameter."))
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/commonAnovaBayesian.R]: occurred* -- Ha, I found something ;)

File: JASP-Engine/JASP/R/commonAnovaBayesian.R [hunk 2]
```diff
@@ -673,85 +771,92 @@
...
+          aesLine <- ggplot2::aes(x = x, y = y, g = g)
+          aesErrorbar <- ggplot2::aes(xmin = xmin, xmax = xmax, y = y, group = g)
+        }
+        
+        
+        maxheight <- min(newymax - dfCri$y[1:min(lInd, 3)])
+        xlab <- nms[i]
+        # ggplot doesn't like our fancy unicode * so we need to remove it
+        # some word hopefully no user every uses
+        keyword <- "myTemporaryMagicReplacementKeyWord"
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/commonAnovaBayesian.R]: Haha -- so escaping it is not enough?

File: JASP-Engine/JASP/R/commonAnovaBayesian.R [hunk 3]
```diff
@@ -1834,11 +1964,43 @@
...
+    samples <- matrix(0, nrow = nIter, ncol = ncol(posterior))
     
+    # get for each parameter which variable they belong to (e.g., 0 and 1 belong to contBinom)
+    stems <- sapply(strsplit(colnames(posterior), "-"), `[`, 1L) # before - is the variable name
+    bases <- c("mu", names(levelInfo[["levelNames"]])) # all options
+    pos <- match(stems, bases) # lookup parameters in variable names
+  
+    # sort the terms inside bases and colnames(effects) so that they match
+    # NOTE: this cannot be done when creating effects because then the colnames don't match with the formulas...
+    bases <- sapply(strsplit(bases, ":"), function(x) paste0(sort(x), collapse = ":"))
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/commonAnovaBayesian.R]: Are these `bases` base64 encoded? Otherwise rogue ":" would be a problem

Comment (vandenman) [reply JASP-Engine/JASP/R/commonAnovaBayesian.R]: These are base64 as long as the formulas and data set given to BayesFactor are.

File: JASP-Engine/JASP/R/commonAnovaBayesian.R (hunk 2 above)
Comment (vandenman) [reply JASP-Engine/JASP/R/commonAnovaBayesian.R]: you're right, that's a much nicer solution

**Pull request merged**